### ISSUE TITLE
Additional info in documentation

### ISSUE THREAD

**Issue by tide34** [jaegertracing/jaeger-operator#1894]

**Comment by tide34** [comment]
I think it's impossible to start jaeger-operator in not cluster-wide mode when follow steps in documention.

**Comment by tide34** [comment]
@yurishkuro hello. Will anyone help solve it?

**Comment by frzifus** [comment]
@rubenvp8510 Do you have experience with it?

**Comment by tide34** [comment]
no one help

**Comment by rubenvp8510** [comment]
Let me try follow your steps, and will back to you with more information, we might need to update the documentation.

**Comment by rubenvp8510** [comment]
Could you attach your manifest yaml file?

**Comment by rubenvp8510** [comment]
Ok I tried to install it on restricted mode (only watching observability namespace) and it works for me.

A couple of points you can check. See if the Roles and RoleBindings are created in the respective namespace. I was able to reproduce the issue with the following command:

`kubectl apply -f manifest.yaml`

But if I do this:
`kubectl apply -f manifest.yaml -n observability`

The error of the namespace goes away, and the operator start successfully.

**Comment by tide34** [comment]
Steps (use minikube):  
1) install cert-manager v1.6.3 (follow [this](https://cert-manager.io/v1.6-docs/installation/kubectl/#steps) guide)  
2) download [jaeger-operator](https://github.com/jaegertracing/jaeger-operator/releases/download/v1.34.0/jaeger-operator.yaml) manifest (from [documentation](https://www.jaegertracing.io/docs/1.34/operator/#installing-the-operator-on-kubernetes))  
3) as documentation told if i want to install jaeger-operator in non cluster wide mode (watch jaeger CR in specific namespaces ONLY, in my case for example namespace app-profile) i need to change the ClusterRole and ClusterBindingRole of the operator manifest to Role and RoleBinding, also set the WATCH_NAMESPACE environmebt variable on the jaeger operator Deployment. Have done it. Got next [manifest](https://github.com/tide34/jaeger-operator/blob/main/jaeger-operator.yaml) 
4) create [this](https://github.com/tide34/jaeger-operator/blob/main/additional.yaml) manifest in namespace app-profile (documentation [link](https://www.jaegertracing.io/docs/1.34/operator/#install-modes))  
  
---  
in log still see line:  
`time="2022-05-29T07:17:46Z" level=info msg="Consider running the operator in a cluster-wide scope for extra features"`  
Pod of jaeger-operator restarts

**Comment by tide34** [comment]
@rubenvp8510 can you try to install jaeger-operator in mode, when loot for jaeger CR in not observability namespace

**Comment by tide34** [comment]
apply manifest from your post above and see in log that jaeger-operator run in cluster wide mode.  
![image](https://user-images.githubusercontent.com/49846209/170861901-2a44f657-1df7-44e0-a37b-b352a51c4f3a.png)

**Comment by rubenvp8510** [comment]
The log line:
`"Consider running the operator in a cluster-wide scope for extra features`

Is normal, it is only indicating that some features are not present on the namespace scoped operator. I'll check your manifest and see if I can reproduce your error.

**Comment by tide34** [comment]
About log ok, i undestand.  
Wait for your answer

**Comment by rubenvp8510** [comment]
I see the problem, Indeed there is a regression here! 

The namespace controller should not be enabled when the operator is not running in cluster wide mode. I'm sending a PR to fix this. hope we can do a minor release this week.

Thanks for reporting.

**Comment by tide34** [comment]
Great, we have found issue. Yhanks for your work. I will look forward to closing PR.  
  
what about that question?  
4. create [this](https://github.com/tide34/jaeger-operator/blob/main/additional.yaml) manifest in namespace app-profile (documentation [link](https://www.jaegertracing.io/docs/1.34/operator/#install-modes). I think that in documentation absent manifest of role jaeger-operator in namespace to watch jaeger CR)  
Can you add in documentaion manifest of jaeger-operator role?

**Comment by rubenvp8510** [comment]
About point 4, wondering why do you need that?  I think the reason is that you are trying to apply the manifest in the observability namespace:  `kubectl apply -f jaeger-operator.yaml -n observability` instead you should apply it on your own namespace: `kubectl apply -f jaeger-operator.yaml -n app-profile`. (But you need to change all references to observability on the manifest)

And may be we need to clarify that on the documentation. Make sense ?

**Comment by tide34** [comment]
i want to deploy jaeger-operator in observability manifest and watch for jaeger CR in app-profile namespace. As documentaion [said](https://www.jaegertracing.io/docs/1.34/operator/#install-modes):  

`It is possible to have the Jaeger Operator running in a given namespace (like, observability) and manage Jaeger resources in another (like, myproject).`  
  
I want to use this mode. After reading documentation i think that only need to create that role in app-profile namespace  
`kind: RoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: jaeger-operator-in-myproject
  namespace: myproject
subjects:
- kind: ServiceAccount
  name: jaeger-operator
  namespace: observability
roleRef:
  kind: Role
  name: jaeger-operator
  apiGroup: rbac.authorization.k8s.io
`  
and nothing more. I got confused what role with name jaeger-operator should be deployed because role it is namespaced resource.  
As i undestand documentation, i have described stepts in guide above what i need to do to deploy jaeger with described scheme of work

**Comment by rubenvp8510** [comment]
Aaa sorry I misunderstood a little bit your scenario, yeah in that case we might want to add a note to the documentation. I'll create the issue and the PR tomorrow :)

**Comment by tide34** [comment]
I will close the issue when check that all work correctly and pod of jaeger-operator don't restarts

**Comment by tide34** [comment]
@rubenvp8510 hello. Please, don't forget create PR

**Comment by rubenvp8510** [comment]
@tide34 I created the first PR: https://github.com/jaegertracing/jaeger-operator/pull/1914 here, I will create a new one on the docu repo to improve it for the use case you mention.

**Comment by tide34** [comment]
@rubenvp8510 Hello. When are you planning to create release with fix?

**Comment by tide34** [comment]
@rubenvp8510 ?

**Comment by tide34** [comment]
Maybe you have mistake in previous message and wrote about version 1.35?  
Can you create a patch release in 1.34.x? I am really looking forward it)

**Comment by tide34** [comment]
Have you decided the release date of 1.35?

**Comment by frzifus** [comment]
Hi @tide34, i will work on 1.35 next week.

**Comment by tide34** [comment]
Hi @frzifus. Please give some info about release date.

**Comment by frzifus** [comment]
probably tomorrow https://github.com/jaegertracing/jaeger-operator/pull/1949

**Comment by tide34** [comment]
Hi, @rubenvp8510. Have you already created issue for documention impruvment? I can't find issue in respository with docs  
![image](https://user-images.githubusercontent.com/49846209/174477927-5ed0f079-3b95-40aa-b0d5-694c72042379.png)

**Comment by eakurdyukov** [comment]
+1  docs looks confusing for me too. Need more examples maybe

**Comment by sereneshikari** [comment]
Yes, I agree the documentation could do a better job of specifying the `Role` spec. In the meantime, I see that the Jaeger team releases a Helm chart for the Jaeger operator (GitHub repo: [helm-charts](https://github.com/jaegertracing/helm-charts)) and they have defined a `Role` Helm template [here](https://github.com/jaegertracing/helm-charts/blob/main/charts/jaeger-operator/templates/role.yaml) which could be useful

**Comment by sereneshikari** [comment]
Hi @ViliusS, if your security posture allows, I suggest running the Jaeger operator in cluster-wide mode itself. There are some issues I observed when running in non-cluster-wide mode that I documented in #2085

**Comment by ViliusS** [comment]
It looks like documentation [here](https://www.jaegertracing.io/docs/1.39/operator/#install-modes) is wrong. It references `jeager-operator` role name however, if Jeager is installed via [static yaml](https://github.com/jaegertracing/jaeger-operator/releases/download/v1.39.0/jaeger-operator.yaml) file the role name is `manager-role`.
